visual-inspection/inspection-backend/inspection/explainer/explain.py

```diff
@@ -49,7 +50,6 @@ def explain(image_file: IO[bytes],
     input_image = Image.open(image_file)
     explainer_input = preprocess(input_image)[0]
 
-    raw_image = EXPLAINERS[method](explainer_input, model_, **settings)
-
+    raw_image = EXPLAINERS[method](explainer_input, model_, index_of_label_to_explain,positive_only_parameter, **settings)
```

Comment:
You can see here why it's problematic to add settings field as additional arguments: All explanation methods now must support `index_of_label_to_explain` and `positive_only_parameter`, which they generally won't.

For this purpose we have the `settings` dictionary that allows arbitrary parameters to be passed.